MapboxDirections/MBDirections.swift

```diff
@@ -130,6 +96,12 @@ open class Directions: NSObject {
     /// The Mapbox access token to associate the request with.
     internal let accessToken: String
     
+    /// The handler that reports the download progress of an offline tile pack
+    internal var offlineProgressHandler: OfflineDownloaderProgressHandler?
+    
+    /// The handler that reports the completion of an offline tile pack
+    internal var offlineCompletionHandler: OfflineDownloaderCompletionHandler?
```

Comment:
It’s a bit awkward that these properties are here in the main class definition because of an extension’s conformance to OfflineDirectionsProtocol.

More significantly, these properties make Directions stateful. Directions originally was stateful, going so far as to store the URLSessionTask for the API request, similar to MapKit’s MKDirections. But that made it impossible for a single Directions object to process multiple requests simultaneously, and it also meant that the object had to be held strongly for the duration of the request. #47 moved Directions to [a stateless singleton model](https://github.com/mapbox/MapboxDirections.swift/blob/master/CHANGELOG.md#v060) that has provided a good deal of flexibility to this library. If we’re going to return to the old model, Directions would need to encapsulate the URLSessionTask objects and provide a method for canceling the request.

Reply:
MapboxDirections.swift takes the approach of exposing the request URL publicly, so you can hook the library into any networking library, not just URLSession but also AFNetworking etc.